module/data/region-behavior/rotate-area.mjs

```diff
@@ -187,11 +173,26 @@ export default class RotateAreaRegionBehaviorType extends foundry.data.regionBeh
       return { _id: wall.id, c: [first.x, first.y, second.x, second.y] };
     });
 
-    // Handle tile and token updates immediately to allow animation
-    await canvas.scene.updateEmbeddedDocuments("Tile", updates.tiles, { dnd5e: { animate: { ...animateFlags } } });
+    // Update status to indicate rotation is occurring and trigger visible animation
+    await this.parent.update(
+      { "system.status": { angle: targetAngle, position, rotating: true } },
+      { dnd5e: { rotateArea: { angle, duration, pivot } } }
+    );
+
+    // Wait for the visible animation to complete before performing document updates
+    const { promise: animationComplete, resolve } = Promise.withResolvers();
+    setTimeout(async () => {
+      resolve(true);
+      await this.parent.update({ "system.status.rotating": false });
+    }, duration);
+    await animationComplete;
+
+    // Update all rotated documents
+    await canvas.scene.updateEmbeddedDocuments("AmbientLight", updates.lights);
+    await canvas.scene.updateEmbeddedDocuments("AmbientSound", updates.sounds);
+    await canvas.scene.updateEmbeddedDocuments("Tile", updates.tiles);
```

Comment:
I guess I missed this earlier, but do we have any guarantee that `canvas.scene` is the scene that this region behaviour is in? Should this be `this.parent.parent` instead?